### PR TITLE
Improve creation of StateMachineFactory, implement #854 

### PR DESCRIPTION
**Pull request by xJoeWoo**

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@xJoeWoo Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-statemachine&pullRequestId=855)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-statemachine&pullRequestId=855) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@xJoeWoo Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-statemachine&pullRequestId=855)!

File: spring-statemachine-core/src/main/java/org/springframework/statemachine/config/StateMachineBuilder.java [hunk 1]
```diff
@@ -112,40 +109,37 @@ public Builder() {
...
+				ObjectStateMachineFactory<S, E> stateMachineFactory = StateMachineFactory.create(builder);
+				ConfigurationData<S, E> stateMachineConfigurationConfig = builder.getOrBuild().stateMachineConfigurationConfig;
 
 				stateMachineFactory.setHandleAutostartup(stateMachineConfigurationConfig.isAutoStart());
 
 				if (stateMachineConfigurationConfig.getBeanFactory() != null) {
 					stateMachineFactory.setBeanFactory(stateMachineConfigurationConfig.getBeanFactory());
 				}
-				return stateMachineFactory.getStateMachine();
+				return stateMachineFactory;
```
Comment (jvalkeal) [inline spring-statemachine-core/src/main/java/org/springframework/statemachine/config/StateMachineBuilder.java:128]: I ran tests(rebased) and `ConfigurationErrorTests` do fail as `MalformedConfigurationException` is not wrapped with `StateMachineException`. I think same try/catch should be in a new `build()` method as well.

**Comment by jvalkeal** [comment]
Polished and merged per 61c58cdee0a55deaa06d399486fdfbeb773fa280

**Pull request closed without merging**